nautobot/docs/models/dcim/deviceredundancygroups.md

```diff
@@ -0,0 +1,250 @@
+```
+
+```json
+{'data': {'devices': [{'name': 'nyc-fw-primary', 'device_redundancy_group': {'name': 'nyc-firewalls', 'members': [{'name': 'nyc-fw-primary', 'device_redundancy_group_priority': 100, 'interfaces': [{'type': 'VIRTUAL', 'name': 'failover-link', 'ip_addresses': [{'host': '172.27.48.0', 'prefix_length': 31}], 'parent_interface': {'name': 'gigabitethernet0/3', 'type': 'A_1000BASE_T'}}]}, {'name': 'nyc-fw-secondary', 'device_redundancy_group_priority': 50, 'interfaces': [{'type': 'VIRTUAL', 'name': 'failover-link', 'ip_addresses': [{'host': '172.27.48.1', 'prefix_length': 31}], 'parent_interface': {'name': 'gigabitethernet0/3', 'type': 'A_1000BASE_T'}}]}]}}]}}
+```
+
+### Retrieving the data - Secondary Failover Unit ("nyc-fw-secondary")
+
+An example data returned from Nautobot is presented below.
+
+```python
+>>> secondary_unit_details = get_gql_failover_details("nyc-fw-secondary").json
+```
+
+```json
+{'data': {'devices': [{'name': 'nyc-fw-secondary', 'device_redundancy_group': {'name': 'nyc-firewalls', 'members': [{'name': 'nyc-fw-primary', 'device_redundancy_group_priority': 100, 'interfaces': [{'type': 'VIRTUAL', 'name': 'failover-link', 'ip_addresses': [{'host': '172.27.48.0', 'prefix_length': 31}], 'parent_interface': {'name': 'gigabitethernet0/3', 'type': 'A_1000BASE_T'}}]}, {'name': 'nyc-fw-secondary', 'device_redundancy_group_priority': 50, 'interfaces': [{'type': 'VIRTUAL', 'name': 'failover-link', 'ip_addresses': [{'host': '172.27.48.1', 'prefix_length': 31}], 'parent_interface': {'name': 'gigabitethernet0/3', 'type': 'A_1000BASE_T'}}]}]}}]}}
+```
+
+### Processing the data - Primary Failover Unit ("nyc-fw-primary")
+
+Before rendering the data, we will process the data to simplify data variables within template:
+
+```python
+>>> hostname = primary_unit_details['data']['devices'][0]['name']
+>>> hostname
+'nyc-fw-primary'
+>>>
+>>> failover_members = primary_unit_details['data']['devices'][0]['device_redundancy_group']['members']
+>>>
+>>> device = [failover_member for failover_member in failover_members if failover_member['name'] == hostname][0]
+>>>
+>>> device
+{'name': 'nyc-fw-primary', 'device_redundancy_group_priority': 100, 'interfaces': [{'type': 'VIRTUAL', 'name': 'failover-link', 'ip_addresses': [{'address': '172.27.48.0/31'}], 'parent_interface': {'name': 'gigabitethernet0/3', 'type': 'A_1000BASE_T'}}]}
+>>>
+>>> failover_peer = [failover_member for failover_member in failover_members if failover_member['name'] != hostname][0]
+>>>
+>>> device_failover_interface = device['interfaces'][0]
+>>> peer_failover_interface = failover_peer['interfaces'][0]
+```
+
+### Creating Cisco ASA Configuration Template - Common for Primary and Secondary Units
+
+Following snippet represents an example Juniper BGP Configuration Template:
+
+```python
+# Configuration Template for Cisco ASA
+template_body="""
+!
+hostname {{ device.name }}
+!
+failover lan unit {{ priority_mapping.device_redundancy_group_priority }}
```

Comment:
Where did `priority_mapping` come from?

Reply:
It is coming via `template.render()` kwargs, although added a few lines about it in the intro 👍